src/components/MAPI/clusters/CreateCluster/CreateClusterControlPlaneNodeAZs.tsx

```diff
@@ -0,0 +1,133 @@
+  AZSelectionZonesUpdater,
+  IUpdateZoneLabelsPayload,
+} from 'Cluster/AZSelection/AZSelectionUtils';
+import { Cluster, ControlPlaneNode } from 'MAPI/types';
+import { determineRandomAZs, getSupportedAvailabilityZones } from 'MAPI/utils';
+import PropTypes from 'prop-types';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { RUMActions } from 'shared/constants/realUserMonitoring';
+import { getProvider } from 'stores/main/selectors';
+import InputGroup from 'UI/Inputs/InputGroup';
+
+import { computeControlPlaneNodesStats } from '../ClusterDetail/utils';
+import {
+  IClusterPropertyProps,
+  withClusterControlPlaneNodeAZs,
+} from './patches';
+
+interface ICreateClusterControlPlaneNodeAZsProps
+  extends IClusterPropertyProps,
+    Omit<
+      React.ComponentPropsWithoutRef<typeof InputGroup>,
+      'onChange' | 'id'
+    > {}
+
+const CreateClusterControlPlaneNodeAZs: React.FC<ICreateClusterControlPlaneNodeAZsProps> = ({
+  id,
+  cluster,
+  controlPlaneNode,
+  onChange,
+  ...props
+}) => {
+  const [azSelector, setAzSelector] = useState(
+    AvailabilityZoneSelection.Automatic
+  );
+
+  const provider = useSelector(getProvider);
+  const azStats = useSelector(getSupportedAvailabilityZones);
+
+  const value = useMemo(() => {
+    return computeControlPlaneNodesStats([controlPlaneNode]).availabilityZones;
+  }, [controlPlaneNode]);
+
+  const [manualZones, setManualZones] = useState<string[]>(value);
+  const [manualZonesIsValid, setManualZonesIsValid] = useState(false);
+
+  const handleZoneChange: AZSelectionZonesUpdater = (azSelection) => (
+    payload
+  ) => {
+    switch (azSelection) {
+      case AvailabilityZoneSelection.Automatic:
+        break;
+
+      case AvailabilityZoneSelection.Manual:
+        setManualZonesIsValid(payload.valid);
+        setManualZones((payload as IUpdateZoneLabelsPayload).zonesArray);
+        break;
+    }
+  };
+
+  const handleChange = (selector: AvailabilityZoneSelection) => {
+    setAzSelector(selector);
+  };
+
+  useEffect(() => {
+    switch (azSelector) {
+      case AvailabilityZoneSelection.Automatic:
+        onChange({
+          isValid: true,
+          patch: withClusterControlPlaneNodeAZs(
+            determineRandomAZs(1, azStats.all)
+          ),
+        });
+        break;
+
+      case AvailabilityZoneSelection.Manual:
+        onChange({
+          isValid: manualZonesIsValid,
+          patch: withClusterControlPlaneNodeAZs(manualZones),
+        });
+        break;
+
+      case AvailabilityZoneSelection.NotSpecified:
+        onChange({
+          isValid: true,
+          patch: withClusterControlPlaneNodeAZs(undefined),
+        });
+        break;
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [azSelector, manualZones, manualZonesIsValid]);
+
+  return (
+    <InputGroup
+      htmlFor={id}
+      label='Control plane node availability zone selection'
```

Comment:
Let's simplify that a bit

```suggestion
      label='Control plane availability zones'
```

Reply:
Yeah, it's pretty long. Should we also change the options from `Automatic` and `Manual` to `Automatic selection` and `Manual selection`?

`Control plane availability zone - Manual` and `Control plane availability zone - Automatic` don't tell much to me

Reply:
Yes, good point, the radio button labels should be

- `Automatic selection`
- `Specify manually`
- `Not specified`

Also in this description text

![image](https://user-images.githubusercontent.com/273727/125250738-e4990d80-e2f6-11eb-8a04-6ebc42078441.png)

The term `availability zone` should be spelled in lower case.